features/telemetry/common/src/main/java/org/opennms/netmgt/telemetry/common/utils/DnsUtils.java

```diff
@@ -0,0 +1,149 @@
+
+    private static String primaryServer = null, secondaryServer = null;
+    private static boolean enable = false;
+    static BundleContext bundleContext;
+
+    static {
+        try {
+            resolver = new ExtendedResolver();
+        } catch (UnknownHostException e) {
+            LOG.debug("Cannot create resolver: {}", e.getMessage());
+        }
+
+        try {
+            bundleContext = FrameworkUtil.getBundle(DnsUtils.class).getBundleContext();
+        } catch (NullPointerException e) {
+            LOG.debug("BundleContext not available: {}", e.getMessage());
+        }
+    }
+
+
+    private static void checkSystemProperties() {
+        if (bundleContext == null) {
+            return;
+        }
+
+        final String primaryServer = bundleContext.getProperty(DNS_PRIMARY_SERVER);
+        final String secondaryServer = bundleContext.getProperty(DNS_SECONDARY_SERVER);
+        final boolean enable = Boolean.parseBoolean(bundleContext.getProperty(DNS_ENABLE));
+
+        if (enable != DnsUtils.enable || !Objects.equals(primaryServer, DnsUtils.primaryServer) || !Objects.equals(secondaryServer, DnsUtils.secondaryServer)) {
+            DnsUtils.enable = enable;
+            DnsUtils.primaryServer = primaryServer;
+            DnsUtils.secondaryServer = secondaryServer;
+            setDnsServers(DnsUtils.primaryServer, DnsUtils.secondaryServer);
+        }
+    }
+
+    public static synchronized void setDnsServers(String... dnsServers) {
+        final String[] notNullDnsServers = (dnsServers == null ? new String[]{} : Arrays.stream(dnsServers)
+                .filter(e -> !Strings.isNullOrEmpty(e))
+                .collect(Collectors.toList())
+                .toArray(new String[]{})
+        );
+
+        try {
+            if (notNullDnsServers.length == 0) {
+                resolver = new ExtendedResolver();
+            } else {
+                resolver = new ExtendedResolver(notNullDnsServers);
+            }
+        } catch (UnknownHostException e) {
+            LOG.debug("Cannot create resolver for given servers {}: {}", dnsServers, e.getMessage());
+        }
+    }
+
+    static ExtendedResolver getResolver() {
+        return resolver;
+    }
+
+    public static Optional<String> reverseLookup(final String inetAddress) {
+        return reverseLookup(InetAddressUtils.addr(inetAddress));
+    }
+
+    public static Optional<String> reverseLookup(final InetAddress inetAddress) {
+        checkSystemProperties();
+
+        if (!DnsUtils.enable) {
+            return Optional.empty();
+        }
+
+        final Lookup lookup = new Lookup(ReverseMap.fromAddress(inetAddress), Type.PTR);
+
+        lookup.setResolver(resolver);
+
+        final Record records[] = lookup.run();
+
+        if (lookup.getResult() == Lookup.SUCCESSFUL) {
+
+            final PTRRecord ptrRecord = (PTRRecord) Arrays.stream(records)
+                    .filter(PTRRecord.class::isInstance)
+                    .findFirst()
```

Comment:
Can we check `records` length and log a WARN or INFO if more than one record was returned?